### PR TITLE
sets rubocop ruby version to 2.1 and apply rubocop -a

### PR DESCRIPTION
**Pull request by raventid**
Let me describe what happened here. I've decided to add `frozen_string_literal` to auto_inject code base. After completing this task I thought that this is one of the rubocop recommendations. This moment a brilliant idea appeard in my mind and I've decided to run `rubocop -a` to fix all possible violations. I think it would be nice to tweak rubocop settings to make it more usable, right now I have a feeling, that it's abandoned.

One thing I'd like to change in these PR - is to use `"` instead of `'` by default. Other things are good.
I personnally use ruby style guide block notation, where you write `do ... end` with multiline blocks, but this code base follows another notation.

What do you think?

### CODE REVIEW

File: lib/dry/auto_inject/dependency_map.rb [hunk 1]
```diff
@@ -33,7 +35,7 @@ def names
       def to_h
         @map.dup.to_h
       end
-      alias_method :to_hash, :to_h
+      alias to_hash to_h
```
Comment (GustavoCaso) [inline lib/dry/auto_inject/dependency_map.rb]: I thought it was better to use `alias_method` than `alias` https://blog.bigbinary.com/2012/01/08/alias-vs-alias-method.html

Comment (raventid) [reply lib/dry/auto_inject/dependency_map.rb]: Hm, it's a good idea for a new spec. I missed that, thanks!

Comment (timriley) [reply lib/dry/auto_inject/dependency_map.rb]: Yeah, I'm not a fan of the lexical scoping of `alias`. Let's leave it as `alias_method`.

File: lib/dry/auto_inject/strategies/args.rb [hunk 2]
```diff
@@ -10,9 +12,9 @@ class Args < Constructor
         def define_new
           class_mod.class_exec(container, dependency_map) do |container, dependency_map|
             define_method :new do |*args|
-              deps = dependency_map.to_h.values.map.with_index { |identifier, i|
+              deps = dependency_map.to_h.values.map.with_index do |identifier, i|
                 args[i] || container[identifier]
-              }
+              end
```
Comment (timriley) [inline lib/dry/auto_inject/strategies/args.rb]: My personal preference is to use the curly braces when the return value of the method accepting the block is significant, which is why I've done that here. I'd rather we didn't change it. In fact, these days I usually just disable the Rubocop rule about this.

File: lib/dry/auto_inject/strategies/args.rb [hunk 3]
```diff
@@ -42,9 +44,9 @@ def initialize(#{initialize_args})
 
         def define_initialize_with_splat(super_method)
           super_params = if super_method.parameters.any? { |type, _| type == :rest }
-            '*args'
-          else
-            "*args[0..#{super_method.parameters.length - 1}]"
+                           '*args'
+                         else
+                           "*args[0..#{super_method.parameters.length - 1}]"
           end
```
Comment (timriley) [inline lib/dry/auto_inject/strategies/args.rb]: Not a great fan of RuboCop's suggestion here either. I'd probably just do it this way if I were rewriting today:

```ruby
super_params =
  if super_method.parameters.any? { |type, _| type == :rest }
    '*args'
  else
    "*args[0..#{super_method.parameters.length - 1}]"
  end
```

File: lib/dry/auto_inject/strategies/hash.rb [hunk 4]
```diff
@@ -10,9 +12,9 @@ class Hash < Constructor
         def define_new
           class_mod.class_exec(container, dependency_map) do |container, dependency_map|
             define_method :new do |options = {}|
-              deps = dependency_map.to_h.each_with_object({}) { |(name, identifier), obj|
+              deps = dependency_map.to_h.each_with_object({}) do |(name, identifier), obj|
                 obj[name] = options[name] || container[identifier]
-              }.merge(options)
+              end.merge(options)
```
Comment (timriley) [inline lib/dry/auto_inject/strategies/hash.rb]: Same, would rather we keep the curly braces.

File: lib/dry/auto_inject/strategies/kwargs.rb [hunk 5]
```diff
@@ -47,17 +49,17 @@ def initialize(#{initialize_params})
         end
 
         def define_initialize_with_splat(super_method)
-          super_kwarg_names = super_method.parameters.each_with_object([]) { |(type, name), names|
-            names << name if [:key, :keyreq].include?(type)
-          }
+          super_kwarg_names = super_method.parameters.each_with_object([]) do |(type, name), names|
+            names << name if %i[key keyreq].include?(type)
```
Comment (timriley) [inline lib/dry/auto_inject/strategies/kwargs.rb]: I don't think that using `%i` here improves readability. In projects lately this is another Rubocop rule that I disable entirely.

File: lib/dry/auto_inject/strategies/kwargs.rb [hunk 6]
```diff
@@ -47,17 +49,17 @@ def initialize(#{initialize_params})
         end
 
         def define_initialize_with_splat(super_method)
-          super_kwarg_names = super_method.parameters.each_with_object([]) { |(type, name), names|
-            names << name if [:key, :keyreq].include?(type)
-          }
+          super_kwarg_names = super_method.parameters.each_with_object([]) do |(type, name), names|
```
Comment (timriley) [inline lib/dry/auto_inject/strategies/kwargs.rb]: Let's keep the curly braces.

File: lib/dry/auto_inject/strategies/kwargs.rb [hunk 7]
```diff
@@ -47,17 +49,17 @@ def initialize(#{initialize_params})
...
+            names << name if %i[key keyreq].include?(type)
+          end
 
           instance_mod.class_exec(dependency_map) do |dependency_map|
             define_method :initialize do |*args, **kwargs|
-              super_kwargs = kwargs.each_with_object({}) { |(key, _), hsh|
+              super_kwargs = kwargs.each_with_object({}) do |(key, _), hsh|
```
Comment (timriley) [inline lib/dry/auto_inject/strategies/kwargs.rb]: Keep the curly braces.

**Comment by GustavoCaso** [comment]
@raventid Thank you so much for taking the time for doing this ❤️ ❤️ ❤️ 
I totally agree with @timriley I think it would be good to split into two PR's.

**Comment by raventid** [comment]
@GustavoCaso @timriley thank you very much for your thorough review.

I totally agree with you, but I have one question. What is the reason to use curly braces for multiline blocks? I'm aware of Weirich's block notation, but never really used it. Any advantages of using this style that I miss?

Plans are:

1) Frozen string literals as separate PR
2) Additions to rubocop.yml according to this review. One thing I'm not sure about is double vs single quotes. I'll stick with `"` in new PR thats not so hard to change anyway.

**Comment by raventid** [comment]
CodeClimate is good, and if no one against this, I'm gonna totally remove rubocop related stuff.

**Comment by solnic** [comment]
@raventid please do that

**Pull request closed without merging**